src/ImageSharp/ApplyProcessors.cs

```diff
@@ -0,0 +1,107 @@
+{
+    using System;
+
+    using ImageSharp.PixelFormats;
+
+    using ImageSharp.Processing;
+
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Mutates the image by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        public static void Mutate<TPixel>(this Image<TPixel> source, Action<IImageOperations<TPixel>> operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+
+            // TODO: add parameter to Configuration to configure how this is created, create an IImageOperationsFactory that cna be used to switch this out with a fake for testing
+            IImageOperations<TPixel> operationsRunner = source.Configuration.ImageOperationsProvider.CreateMutator(source);
+            operations(operationsRunner);
+        }
+
+        /// <summary>
+        /// Mutates the image by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        public static void Mutate<TPixel>(this Image<TPixel> source, params IImageProcessor<TPixel>[] operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+
+            // TODO: add parameter to Configuration to configure how this is created, create an IImageOperationsFactory that cna be used to switch this out with a fake for testing
+            IImageOperations<TPixel> operationsRunner = source.Configuration.ImageOperationsProvider.CreateMutator(source);
+            operationsRunner.ApplyProcessors(operations);
+        }
+
+        /// <summary>
+        /// Clones the current image mutating the clone by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        /// <returns>Anew Image which has teh data from the <paramref name="source"/> but with the <paramref name="operations"/> applied.</returns>
+        public static Image<TPixel> Clone<TPixel>(this Image<TPixel> source, Action<IImageOperations<TPixel>> operations)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Guard.NotNull(operations, nameof(operations));
+            var generated = new Image<TPixel>(source);
```

Comment:
I was planning to create a new internal interface `ICloneableImageProcessor<T> : IImageProcessor<TPixel>` (or similare ) then in the `DefaultImageOperator` check if the passed in `IImageProcessor` is in fact an `ICloneableImageProcessor` and if it is then instead of calling `IImageProcessor.Apply()` then it instead call `IImageProcessor.CloneAndApply()` which returns the cloned image. All it would require is the `DefaultImageOperator` cloning the source image lazerly the first time we encounter an `IImageProcesssor` and its not an `ICloneableImageProcessor` and if it is on then it will retain the cloned image and just call Apply with it on all remaining `IImageProcesssor`